src/System.Private.CoreLib/src/System/Delegate.cs

```diff
@@ -15,11 +16,11 @@ namespace System
     public abstract class Delegate : ICloneable, ISerializable
     {
         // _target is the object we will invoke on
-        internal object _target;
+        internal object _target = null!; // Initialized by VM as needed
```

Comment:
We do not initialize fields to null anywhere else. Does the nullable work going to mean that we will need to explicitly initialize fields to null? It does not look like an improvement.

Reply:
This particular case should go away as it should be declared as `object?` instead of `object`.

But in the general case, yeah, this is an issue.  If you declare something as non-nullable in a class and then don't initialize it in the ctor, the compiler complains.  The three options at that point are either:
- Make the field nullable
- Initialize it, which generally in these cases means `null`, and since it's declared non-nullable, `null!` to suppress the warning.
- Disable nullability warnings around the relevant ctor(s)

The workaround being planned right now is for the compiler to not emit IL for initializing a field to its default value as part of the declaration, which seems like a worthwhile micro-optimization to do even without the nullable work (I actually thought the compiler already did that).

Reply:
> Initialize it, which generally in these cases means null

When would you want to do this instead of marking the field nullable?

Reply:
> When would you want to do this instead of marking the field nullable?

e.g. if the initialization is done by a helper the ctor calls, such that the field shouldn't ever be null after the ctor completes.